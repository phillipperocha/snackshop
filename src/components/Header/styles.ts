import styled, { css } from 'styled-components'

type WrapperProps = { hasBanner: boolean }
export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, hasBanner }) => css`
    padding-top: 0;
    background: ${theme.colors.primary};
    ${hasBanner && { paddingTop: '5px' }}
  `}
`

export const Banner = styled.div`
  ${({ theme }) => css`
    display: flex;
    background-image: url('/assets/img/bannerBackground.svg');
    background-size: cover;
    align-items: center;
    padding: 0 ${theme.spacings.xsmall};
    gap: ${theme.spacings.xxsmall};
    min-height: 10.6rem;

    span {
      font-size: ${theme.font.sizes.xlarge};
    }
  `}
`

export const BannerImage = styled.div`
  ${({ theme }) => css`
    display: flex;
    width: 11.4rem;
    justify-content: flex-end;
    margin-left: -${theme.spacings.xsmall};
  `}
`

export const HeaderText = styled.div`
  ${({ theme }) => css`
    display: flex;
    min-height: 6.3rem;
    padding: ${theme.spacings.xsmall};
    align-items: center;

    svg {
      cursor: pointer;
      margin-right: ${theme.spacings.xxsmall};
    }

    h3 {
      color: ${theme.colors.white};
      font-weight: bold;
      font-size: ${theme.font.sizes.large};
      display: inline;
    }
  `}
`
