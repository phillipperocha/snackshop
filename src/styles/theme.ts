export default {
  border: {
    radius: '5px',
  },
  font: {
    family: "'Roboto', sans-serif",
    weight: {
      normal: 400,
      bold: 700,
      black: 900,
    },
    sizes: {
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '3.0rem',
    },
  },
  colors: {
    primary: '#FF0000',
    mainBg: '#FFFFFF',
    white: '#FFFFFF',
    black: '#333333',
    lightGray: '#DDDDDD',
    gray: '#AEAEAE',
    darkGray: '#707070',
  },
  spacings: {
    xxsmall: '1.0rem',
    xsmall: '2.0rem',
    small: '2.4rem',
    medium: '3.5rem',
  },
}
