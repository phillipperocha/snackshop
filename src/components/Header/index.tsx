import * as S from './styles'
import { Link } from 'react-router-dom'

import { ReturnArrowIcon } from 'components/Icons'
import theme from 'styles/theme'

export type HeaderProps = {
  returnHref?: string
  children: React.ReactNode
  bannerText?: React.ReactNode
  bannerImage?: string
}

const Header = ({
  children,
  returnHref,
  bannerText,
  bannerImage,
}: HeaderProps) => (
  <S.Wrapper hasBanner={!!bannerText}>
    {bannerText && (
      <S.Banner>
        {bannerImage && (
          <S.BannerImage>
            <img src={bannerImage} alt="Imagem de coxinhas" />
          </S.BannerImage>
        )}
        <span>{bannerText}</span>
      </S.Banner>
    )}
    <S.HeaderText>
      {!!returnHref && (
        <Link to={returnHref}>
          <ReturnArrowIcon color={theme.colors.white} size="20px" />
        </Link>
      )}
      <h3>{children}</h3>
    </S.HeaderText>
  </S.Wrapper>
)

export default Header
