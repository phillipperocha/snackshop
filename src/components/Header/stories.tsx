import { Story, Meta } from '@storybook/react/types-6-0'

import Header from '.'

export default {
  title: 'Header',
  component: Header,
  argTypes: {
    children: {
      type: 'string',
    },
  },
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

export const Default: Story = ({ children, ...args }) => (
  <Header {...args}>{children}</Header>
)

Default.args = {
  children: 'Monte seu cento de salgado (1, 2 ou 4 opções)',
}

export const WithBanner: Story = ({ children, ...args }) => (
  <Header {...args}>{children}</Header>
)

WithBanner.args = {
  children: 'Monte seu cento de salgado (1, 2 ou 4 opções)',
  withBanner: true,
  bannerText: (
    <span>
      ESCOLHEU <strong>COXINHA</strong> GANHOU DESCONTO DE <strong>10%</strong>{' '}
      NO SEU CENTO DE SALGADOS
    </span>
  ),
}
