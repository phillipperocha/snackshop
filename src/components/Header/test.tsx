import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import { MemoryRouter } from 'react-router'
import theme from 'styles/theme'

import Header from '.'

describe('<Header />', () => {
  it('should render the default header correctly', () => {
    const { container } = renderWithTheme(<Header>Carrinho</Header>)

    const heading = screen.getByRole('heading', { name: /carrinho/i })

    expect(heading).toBeInTheDocument()

    expect(heading).toHaveStyle({
      color: theme.colors.white,
      'font-weight': 'bold',
    })

    expect(heading.parentElement).toHaveStyle({
      'min-height': '6.3rem',
      display: 'flex',
    })

    expect(container.firstChild).toHaveStyle({
      background: theme.colors.primary,
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the header with banner', () => {
    const props = {
      children: 'Monte seu cento de salgado (1, 2 ou 4 opções)',
      bannerText: 'Promoção de coxinha',
      bannerImage: '/img/coxinha.png',
    }

    const { container } = renderWithTheme(<Header {...props} />)

    const heading = screen.getByRole('heading', { name: props.children })

    expect(heading).toBeInTheDocument()

    expect(heading).toHaveStyle({
      color: theme.colors.white,
      'font-weight': 'bold',
    })

    const bannerText = screen.getByText(props.bannerText)

    expect(bannerText).toBeInTheDocument()

    expect(bannerText.parentElement).toHaveStyle({
      display: 'flex',
      'background-image': 'url(/img/bannerBackground.svg)',
      'background-size': 'cover',
      'min-height': '10.6rem',
    })

    expect(
      screen.getByRole('img', { name: /imagem de coxinhas/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toHaveStyle({
      background: theme.colors.primary,
      'padding-top': '5px',
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render header with icon', () => {
    renderWithTheme(
      <MemoryRouter initialEntries={['/']}>
        <Header returnHref="/">Carrinho</Header>
      </MemoryRouter>
    )

    const icon = screen.getByLabelText(/return arrow icon/i)

    expect(icon).toBeInTheDocument()

    expect(icon).toHaveStyle({
      color: theme.colors.white,
    })

    expect(screen.getByRole('link').getAttribute('href')).toBe('/')
  })
})
