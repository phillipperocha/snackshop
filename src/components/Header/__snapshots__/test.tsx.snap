// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the default header correctly 1`] = `
.c0 {
  padding-top: 0;
  background: #FF0000;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 6.3rem;
  padding: 2.0rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 svg {
  cursor: pointer;
  margin-right: 1.0rem;
}

.c1 h3 {
  color: #FFFFFF;
  font-weight: bold;
  font-size: 1.8rem;
  display: inline;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <h3>
      Carrinho
    </h3>
  </div>
</div>
`;

exports[`<Header /> should render the header with banner 1`] = `
.c0 {
  padding-top: 0;
  background: #FF0000;
  padding-top: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-image: url('/img/bannerBackground.svg');
  background-size: cover;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 2.0rem;
  gap: 1.0rem;
  min-height: 10.6rem;
}

.c1 span {
  font-size: 2.0rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 11.4rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  margin-left: -2.0rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 6.3rem;
  padding: 2.0rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 svg {
  cursor: pointer;
  margin-right: 1.0rem;
}

.c3 h3 {
  color: #FFFFFF;
  font-weight: bold;
  font-size: 1.8rem;
  display: inline;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <img
        alt="Imagem de coxinhas"
        src="/img/coxinha.png"
      />
    </div>
    <span>
      Promoção de coxinha
    </span>
  </div>
  <div
    class="c3"
  >
    <h3>
      Monte seu cento de salgado (1, 2 ou 4 opções)
    </h3>
  </div>
</div>
`;
